<!doctype html>
<html class="no-js">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><link rel="index" title="Index" href="../../genindex.html" /><link rel="search" title="Search" href="../../search.html" />

    <meta name="generator" content="sphinx-3.5.4, furo 2021.04.11.beta34"/>
        <title>Brief overview of machine learning in computational biology - SfB knowledge base documentation</title>
      <link rel="stylesheet" href="../../_static/styles/furo.css?digest=59ab60ac09ea94ccfe6deddff6d715cce948a6fc">
    <link rel="stylesheet" href="../../_static/pygments.css">
    <link media="(prefers-color-scheme: dark)" rel="stylesheet" href="../../_static/pygments_dark.css">
    


<style>
  :root {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  
  }
  @media (prefers-color-scheme: dark) {
    :root {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
  }

  /* For allowing end-user-specific overrides */
  .override-light {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  
  }
  .override-dark {
    --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
  }
</style><link rel="stylesheet" href="../../_static/styles/furo-extensions.css?digest=d391b54134226e4196576da3bdb6dddb7e05ba2b"></head>
  <body dir="">
    
<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none"
      stroke-width="1.5" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round">
      <path stroke="none" d="M0 0h24v24H0z"/>
      <line x1="4" y1="6" x2="20" y2="6" />
      <line x1="10" y1="12" x2="20" y2="12" />
      <line x1="6" y1="18" x2="20" y2="18" />
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none"
      stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"
      class="feather feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none"
      stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"
      class="feather feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation"></label>
<label class="overlay toc-overlay" for="__toc"></label>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="../../index.html"><div class="brand">SfB knowledge base  documentation</div></a>
    </div>
    <div class="header-right">
      <label class="toc-overlay-icon toc-header-icon" for="__toc">
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand" href="../../index.html">
  
  
  <span class="sidebar-brand-text">SfB knowledge base  documentation</span>
  
</a><form class="sidebar-search-container" method="get" action="../../search.html">
  <input class="sidebar-search" placeholder=Search name="q">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form><div class="sidebar-scroll"><div class="sidebar-tree">
  <p class="caption"><span class="caption-text">Getting started:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../getting_started_bioinf.html">Getting started in bioinformatics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../biology/genomics_intro.html">Getting started with genomics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../biology/immunology_intro.html">Getting started with immunology and adaptive immune receptors</a></li>
</ul>
<p class="caption"><span class="caption-text">Topics:</span></p>
<ul>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../analysis_design.html">Designing an analysis</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/><label for="toctree-checkbox-1"><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../statistics.html">Statistics: reference</a></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../ml_and_causality.html">Machine learning and causal inference</a><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/><label for="toctree-checkbox-2"><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul class="simple">
</ul>
</li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../specialized_topics_in_ml_and_stats.html">Specialized topics in machine learning and statistics</a><input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/><label for="toctree-checkbox-3"><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l3"><a class="reference internal" href="../specialized_ml_topics/DL_book_ch5_1_overview.html">Machine Learning Basics</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../biology/genomics.html">Genomics: references</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../biology/immunology.html">Immunology: references</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../programming.html">Programming</a><input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/><label for="toctree-checkbox-4"><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../clean_code.html">Clean code</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../../setting_up_project.html">Setting up and organizing a project</a><input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/><label for="toctree-checkbox-5"><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../sharing_tools.html">Services to share developed tools</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../running_analysis.html">Running an analysis</a></li>
</ul>

</div>
</div>
      </div>
      
    </div>
  </aside>
  <main class="main">
    <div class="content">
      <article role="main">
        <label class="toc-overlay-icon toc-content-icon" for="__toc">
          <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
        </label>
        <div class="section" id="brief-overview-of-machine-learning-in-computational-biology">
<h1>Brief overview of machine learning in computational biology<a class="headerlink" href="#brief-overview-of-machine-learning-in-computational-biology" title="Permalink to this headline">Â¶</a></h1>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Aim: get an intuition and high-level understanding of what machine learning is and types of problems it can help solving from a
computational biology perspective</p>
<p>Level: beginner ðŸŒ±</p>
</div>
<p>With the development of sequencing technologies, large amounts of data became available that can be examined for biological properties.
For example, from <a class="reference external" href="https://vdjdb.cdr3.net/">the VDJdb database</a> (<a class="reference external" href="https://academic.oup.com/nar/article/48/D1/D1057/5582255">Bagaev et al. 2020</a>),
we can retrieve T-cell receptor sequences specific to the epitope YLQPRTFLL of SARS-CoV-2 and discover motifs in these amino acid sequences with or
without genetic background, as shown below (the motifs and the sequences were obtained from VDJdb):</p>
<img alt="This figure shows a table with a list of immune receptor (TCR) sequences and the motifs that were extracted as common in those sequences when adjusted and when not adjusted for the genetic background." src="../../_images/TCR_motif_example.png"/>
<p>One way we can approach the analysis of these sequences is to build a position weight matrices showing the product multinomial distribution of amino
acids. In this way we can describe these sequences. On the other hand, if we want to predict for a new sequence if it is specific to a virus or not,
we might need a bit different approach.</p>
<div class="section" id="machine-learning-is-a-powerful-approach-to-discovering-patterns-in-biological-data">
<h2>Machine learning is a powerful approach to discovering patterns in (biological) data<a class="headerlink" href="#machine-learning-is-a-powerful-approach-to-discovering-patterns-in-biological-data" title="Permalink to this headline">Â¶</a></h2>
<p>Machine learning (and statistics) includes a set of methods that allow for making inference about the data. For example if we want to predict if a
receptor sequence will bind the virus or not, we could fit a logistic regression model on the receptor sequence data and then predict the binding for
the new receptors using the model.</p>
<img alt="This figure shows a simple machine learning workflow including raw labeled data that is transformed to encoded data, and then provided to the ML model fitted via a training procedure. When new raw data is encoded, it is provided to the trained model to make predictions." src="../../_images/simple_ML_workflow.png"/>
<p>Some applications of machine learning in biology include predicting transcription factor binding, antigen binding, translation initiation site
discovery, splicing prediction, single-cell RNA-seq clustering, variant calling.</p>
<p>For instance, in antigen binding prediction, immune receptors
(proteins) bind to antigens (e.g., parts of viruses) to help eliminate them. The machine learning helps then solve the classification problem of
predicting if a new receptor will bind to an antigen given a set of receptors that are known to bind that antigen (e.g., <a class="reference external" href="https://www.sciencedirect.com/science/article/pii/S2211124721001704">Akbar et al. 2021</a>).</p>
<p>Transcription factors are proteins that bind to certain sites in DNA and regulate transcription of genes, so the corresponding machine learning
problem can be formulated as follows: given a set of DNA sequences that we know if the transcription factor will bind or not, can we predict if the
same transcription factor will bind to a new DNA sequence (e.g., <a class="reference external" href="https://www.nature.com/articles/nbt.3300">Alipanahi et al. 2015</a>).</p>
<p>In splicing scenario, where splicing refers to processing of precursor RNA that creates messenger RNA by removing
non-coding regions (introns) and connects gene-coding regions (exons) together, machine learning has been used to predict how often exons will be
included in the transcripts in a regression problem (e.g., <a class="reference external" href="https://science.sciencemag.org/content/347/6218/1254806.abstract">Xiong, Alipanahi and Lee et al. 2015</a>).</p>
<p>Single-cell RNA-seq clustering can be performed to idenfity new cell types by applying a dimensionality reduction technique on cell count data,
followed by clustering the data, e.g., by using k-means algorithm (e.g., <a class="reference external" href="https://www.nature.com/articles/s41576-018-0088-9">Kiselev et al. 2019</a>).</p>
<img alt="This figure shows a plot with principal components 1 and 2 on x and y axis respectively, with points in clusters. When a clustering algorithm is applied, the same plot is shown but the points are grouped into clusters of different colors, and one cluster is labeled to represent a new cell type." src="../../_images/clustering_rna_seq.png"/>
</div>
<div class="section" id="computational-biology-poses-unique-challenges-for-machine-learning">
<h2>Computational biology poses unique challenges for machine learning<a class="headerlink" href="#computational-biology-poses-unique-challenges-for-machine-learning" title="Permalink to this headline">Â¶</a></h2>
<p>The unique challenges of computational biology for machine learning come from:</p>
<ul class="simple">
<li><p>dimensionality and the dataset size (e.g., immune repertoire data includes over 100 thousand receptors per individual),</p></li>
<li><p>signal and noise in the data (e.g., in immune repertoire field, only very few of hundreds of thousands of receptors per individual is specific to one disease that we want to predict),</p></li>
<li><p>the ground truth is unknown and the datasets might be weakly labeled (e.g., we know that an immune repertoire belongs to a person with the immune disease in question, but we do not know which receptors are specific for that disease),</p></li>
<li><p>there might be selection bias, meaning that there is some preferential selection of data points into the study that might obscure the existing relationships or introduce non-existing ones.</p></li>
</ul>
</div>
<div class="section" id="recommended-further-reading">
<h2>Recommended further reading<a class="headerlink" href="#recommended-further-reading" title="Permalink to this headline">Â¶</a></h2>
<ul class="simple">
<li><p>Wainberg M, Merico D, Delong A, Frey BJ. Deep learning in biomedicine. Nature Biotechnology. 2018;36(9):829-838. doi:10.1038/nbt.4233</p></li>
<li><p>Leung MKK, Delong A, Alipanahi B, Frey BJ. Machine Learning in Genomic Medicine: A Review of Computational Problems and Data Sets. Proceedings of the IEEE. 2016;104(1):176-197. doi:10.1109/JPROC.2015.2494198</p></li>
</ul>
</div>
</div>

      </article>
      <footer>
        
        <div class="related-pages">
          
          
        </div>

        <div class="related-information">
              Copyright &#169; 2021, SfB_UiO
            |
            Built with <a href="https://www.sphinx-doc.org/">Sphinx</a>
              and
              <a class="muted-link" href="https://pradyunsg.me">@pradyunsg</a>'s
              <a href="https://github.com/pradyunsg/furo">Furo theme</a>.
            |
            <a class="muted-link" href="../../_sources/computational/ml_into/overview.rst.txt"
               rel="nofollow">
              Show Source
            </a>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer">
      
      
      <div class="toc-sticky toc-scroll">
        <div class="toc-title-container">
          <span class="toc-title">
            Contents
          </span>
        </div>
        <div class="toc-tree-container">
          <div class="toc-tree">
            <ul>
<li><a class="reference internal" href="#">Brief overview of machine learning in computational biology</a><ul>
<li><a class="reference internal" href="#machine-learning-is-a-powerful-approach-to-discovering-patterns-in-biological-data">Machine learning is a powerful approach to discovering patterns in (biological) data</a></li>
<li><a class="reference internal" href="#computational-biology-poses-unique-challenges-for-machine-learning">Computational biology poses unique challenges for machine learning</a></li>
<li><a class="reference internal" href="#recommended-further-reading">Recommended further reading</a></li>
</ul>
</li>
</ul>

          </div>
        </div>
      </div>
      
      
    </aside>
  </main>
</div>
    <script id="documentation_options" data-url_root="../../" src="../../_static/documentation_options.js"></script>
    <script src="../../_static/jquery.js"></script>
    <script src="../../_static/underscore.js"></script>
    <script src="../../_static/doctools.js"></script>
    <script src="../../_static/scripts/main.js?digest=e931d09b2a40c1bb82b542effe772014573baf67"></script></body>
</html>